resolves #5 add SVGs for icon-sized logo

mojavelinux opened the pull request:
I modified the SVGs as follows:

* add 6px stroke to lines using the appropriate color
* dropped the bottom bar on the A down by 4px

Here's how the SVGs should be distilled into icon-sized PNGs

* open SVG in gimp at 600dpi
* scale to 32x32 or 16x16 using Sinc scaling
* for the 16x16 icon, sharpen using a factor of 30

mojavelinux committed:
resolves #5 add SVGs for icon-sized logo

- add SVGs with thicker stroke for making icon-sized logos
- add script to export icons to PNG and produce a favicon
- add outline to filename for outline logos

mojavelinux commented:
@Mogztter Do you think we should put the PNGs in the repository? Perhaps just for the icon size? Or perhaps we can automate it?

ggrossetie commented:
> Do you think we should put the PNGs in the repository? Perhaps just for the icon size? Or perhaps we can automate it?

I think we should versioned it.
If i'm not mistaken GitHub can create a visual diff ? If it's the case it should be easier to spot improvements (when we update the source SVG).

And yes we should have a automatic conversion from SVG to PNG for icon size 👍

mojavelinux commented:
Okay, let's go ahead and put the PNGs in there for the icon size at least (since it requires special steps for saving).

I also think we should move these files into the icon directory so they aren't confused with the main logos.

> I think we should versioned it.

We can use tags to handle that part.

mojavelinux commented:
I added a script to make the PNGs for the icons (we could expand this script in the future to make PNGs for everything).

I also moved the icon SVGs into the logo/icon folder so they don't get confused with the regular logo.

ggrossetie reviewed (approved):
👍

mojavelinux commented:
Thanks for reviewing. I think I'm going to extend the script to make all the PNG files and the favicon file. I'm still thinking about pushing those into a tag so they are versioned, but that might come later.

ggrossetie commented on export-to-png.sh:
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/bash
```
@mojavelinux On my machine (Ubuntu 18.04 LTS), the path should be `#!/bin/bash` otherwise I get a `zsh: ./export-to-png.sh: bad interpreter: /usr/bin/bash`

mojavelinux replied:
Thanks for the heads up. I'll update it.

ggrossetie commented:
I had a change to give it a try and there's a few issues on my machine.
There's a white background on the png images:

![asciidoctor-logo-fill-color-64](https://user-images.githubusercontent.com/333276/47137506-0fc1b000-d2b7-11e8-9c62-5110e2bf4b34.png)

I tried to add `-transparent white` and/or `-background rgba(0,0,0,0)` (as mentioned in https://stackoverflow.com/questions/9155377/set-transparent-background-using-imagemagick-and-commandline-prompt) but it's not working as expected.
In the first case the A (which is also white) will be transparent and in the second case the logo still contains white lines:

![asciidoctor-logo-fill-color-64](https://user-images.githubusercontent.com/333276/47137706-b0b06b00-d2b7-11e8-98db-322b9a6df3db.png)

Also `asciidoctor-logo-fill-color` and `asciidoctor-logo-fill-bw` contain a black border around the letter A:

![asciidoctor-logo-fill-color](https://user-images.githubusercontent.com/333276/47137507-0fc1b000-d2b7-11e8-9321-718d7eb54a5b.png)

![asciidoctor-logo-fill-bw](https://user-images.githubusercontent.com/333276/47137508-0fc1b000-d2b7-11e8-907a-02aee69518d5.png)


#### Version

```
convert --version
Version: ImageMagick 6.9.7-4 Q16 x86_64 20170114 http://www.imagemagick.org
Copyright: © 1999-2017 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP 
Delegates (built-in): bzlib djvu fftw fontconfig freetype jbig jng jpeg lcms lqr ltdl lzma openexr pangocairo png tiff wmf x xml zlib
```

ggrossetie commented:
With the latest version of ImageMagick, the A letter disappeared:
```
convert --version
```
```
Version: ImageMagick 7.0.8-12 Q16 x86_64 2018-10-18 https://imagemagick.org
Copyright: © 1999-2018 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP 
Delegates (built-in): bzlib djvu fftw fontconfig freetype jbig jng jpeg lcms lqr lzma openexr pangocairo png tiff wmf x xml zlib
```
```
$ ./export-to-png.sh
```
```
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
convert: non-conforming drawing primitive definition `normal' @ error/draw.c/RenderMVGContent/3052.
convert: non-conforming drawing primitive definition `letter-spacing' @ error/draw.c/RenderMVGContent/4348.
```

![asciidoctor-logo-fill-color-64](https://user-images.githubusercontent.com/333276/47138461-be66f000-d2b9-11e8-8b7d-f52708fd038d.png)

mojavelinux commented:
After **a lot** of searching and banging head on desk, I finally realized that the problem is the order of arguments. If you put the image to convert before the other arguments, you always get a white background. The the input and output files are the final two arguments, then `-background none` works.

Seriously, why does this have to be so difficult :scream_cat:

mojavelinux commented:
I've updated the script.

ggrossetie commented:
Thanks!
It's at the very least confusing... 😕
